runner/ollamarunner/runner.go

```diff
@@ -432,14 +426,14 @@ func (s *Server) processBatch() error {
 		}
 
 		// sample a token
-		vocabSize := len(f32s) / len(options.Outputs)
-		tokens, err := sample.Sample(logits[seq.iBatch*vocabSize:(seq.iBatch+1)*vocabSize], seq.samplers...)
+		vocabSize := len(logits) / len(options.Outputs)
+
+		sampledToken, err := seq.sampler.Sample(logits[seq.iBatch*vocabSize : (seq.iBatch+1)*vocabSize])
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to sample token: %w", err)
 		}
 
-		// TODO(jessegross): Sampler will output a single int32 in the future
-		token := int32(tokens[0])
+		token := sampledToken
```

Comment:
`token` and `sampledToken` are now the same thing.

Reply:
Oops yes thank you for the catch 🙏🏽